plugins/catalog-react/src/components/UserListPicker/UserListPicker.test.tsx

```diff
@@ -306,76 +376,220 @@ describe('<UserListPicker />', () => {
+                string,
+                string
+              >
+            )['relations.ownedBy']
+          ) {
+            return new Promise(() => {});
+          }
+          return mockQueryEntitiesImplementation(request);
+        });
 
-      it('does not reset the filter while entities are loading', () => {
-        render(picker({ loading: true }));
+        render(
+          <Picker initialFilter="owned" />,
+        ); /*  picker({ loading: true })*/
 
+        await waitFor(() =>
+          expect(mockCatalogApi.queryEntities).toHaveBeenCalledTimes(3),
+        );
         expect(updateFilters).not.toHaveBeenCalledWith({
-          user: new UserListFilter(
-            'all',
-            mockIsOwnedEntity,
-            mockIsStarredEntity,
-          ),
+          user: EntityUserListFilter.all(),
         });
       });
 
-      it('does not reset the filter when entities are loaded', () => {
-        render(picker({ loading: false }));
+      it('does not reset the filter when entities are loaded', async () => {
+        render(<Picker initialFilter="owned" />);
+
+        await waitFor(() =>
+          expect(mockCatalogApi.queryEntities).toHaveBeenCalledTimes(3),
+        );
 
-        expect(updateFilters).toHaveBeenLastCalledWith({
-          user: new UserListFilter(
-            type,
-            mockIsOwnedEntity,
-            mockIsStarredEntity,
-          ),
+        await waitFor(() =>
+          expect(updateFilters).toHaveBeenLastCalledWith({
+            user: EntityUserListFilter.owned(expect.any(Array)),
+          }),
+        );
+      });
+    });
+
+    describe(`when there are some starred entities present`, () => {
+      it('does not reset the filter while entities are loading', async () => {
+        mockCatalogApi.queryEntities?.mockImplementation(request => {
+          if (
+            (
+              (request as QueryEntitiesInitialRequest).filter as Record<
+                string,
+                string
+              >
+            )['metadata.name']
+          ) {
+            return new Promise(() => {});
+          }
+          return mockQueryEntitiesImplementation(request);
         });
+
+        render(
+          <Picker initialFilter="starred" />,
+        ); /*  picker({ loading: true })*/
+
+        await waitFor(() =>
+          expect(mockCatalogApi.queryEntities).toHaveBeenCalledTimes(3),
+        );
+        expect(updateFilters).not.toHaveBeenCalledWith({
+          user: EntityUserListFilter.all(),
+        });
+      });
+
+      it('does not reset the filter when entities are loaded', async () => {
+        render(<Picker initialFilter="starred" />);
+
+        await waitFor(() =>
+          expect(mockCatalogApi.queryEntities).toHaveBeenCalledTimes(3),
+        );
+
+        await waitFor(() =>
+          expect(updateFilters).toHaveBeenLastCalledWith({
+            user: EntityUserListFilter.starred(expect.any(Array)),
```

Comment:
could expect the actual array right? maybe there's no good syntax for that

Reply:
you're right! 😄